### ISSUE TITLE
Script execution terminates before all operators complete

### ISSUE THREAD

**Issue by pditommaso** [nextflow-io/nextflow#259]
Consider the following snippet: 

```
Channel 
    .from('alpha', 'beta', 'gamma') 
    .collectFile(name: 'sample.txt', newLine: true) 
    .subscribe { 
        println "Entries are saved to file: $it" 
        println "File content is: ${it.text}" 
    } 

println 123 
```

It prints `123` and exit without awaiting the completion of the `collectFile` operator.

**Comment by pditommaso** [comment]
This is a regression introduced in version `0.22.5` by this commit https://github.com/nextflow-io/nextflow/commit/76dd0ea0be6f60ab907040498971319a0872bb41

**Comment by pditommaso** [comment]
Patch https://github.com/nextflow-io/nextflow/commit/01aebeb4aabcce4004e877a6c2df867bf856dfec is invalid.

**Comment by pditommaso** [comment]
Fix included in version `0.22.6`
